azurerm/resource_arm_api_management_subscription_test.go

```diff
@@ -0,0 +1,298 @@
+					resource.TestCheckResourceAttr(resourceName, "state", "submitted"),
+					resource.TestCheckResourceAttrSet(resourceName, "subscription_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "primary_key"),
+					resource.TestCheckResourceAttrSet(resourceName, "secondary_key"),
+				),
+			},
+			{
+				Config: testAccAzureRMAPIManagementSubscription_update(ri, location, "active"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAPIManagementSubscriptionExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "state", "active"),
+				),
+			},
+			{
+				Config: testAccAzureRMAPIManagementSubscription_update(ri, location, "suspended"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAPIManagementSubscriptionExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "state", "suspended"),
+				),
+			},
+			{
+				Config: testAccAzureRMAPIManagementSubscription_update(ri, location, "cancelled"),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAPIManagementSubscriptionExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "state", "cancelled"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMAPIManagementSubscription_complete(t *testing.T) {
+	resourceName := "azurerm_api_management_subscription.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMAPIManagementSubscriptionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMAPIManagementSubscription_complete(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAPIManagementSubscriptionExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "state", "active"),
+					resource.TestCheckResourceAttrSet(resourceName, "subscription_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "primary_key"),
+					resource.TestCheckResourceAttrSet(resourceName, "secondary_key"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testCheckAzureRMAPIManagementSubscriptionDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*ArmClient).apiManagementSubscriptionsClient
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_api_management_subscription" {
+			continue
+		}
+
+		subscriptionId := rs.Primary.Attributes["subscription_id"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		serviceName := rs.Primary.Attributes["api_management_name"]
+
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+		resp, err := client.Get(ctx, resourceGroup, serviceName, subscriptionId)
+
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return err
+			}
+		}
+
+		return nil
```

Comment:
Should this be removed so it loops through all `azurerm_api_management_subscription` resources in the state to verify they have been deleted? I really wish we had a provider linter to catch these 😄

Reply:
it shouldn't matter since the test cases only allow for one to be successfully provisioned, so I don't think that's necessary? (e.g. we could remove this line but that doesn't change the test expectation so it's fine?)